docs/enhancements/image-service-cloud-authentication.md

```diff
@@ -0,0 +1,164 @@
+
+A new API endpoint will be added:
+
+* `/v2/infra-envs/{infra_env_id}/downloads/refresh-image-secret`
+
+This new endpoint and the infraEnv GET/PATCH/POST will be protected by user credential
+authentication and authorization.
+
+Both the assisted service and image service will accept a new URL parameter `image_secret`
+to be used with this authentication method.
+
+### Risks and Mitigations
+
+Validating authentication against a database has a performance impact compared to JWT
+validation which doesn't need to contact a separate system. That said, we still already
+check the database for authorization and the added time to check that the correct secret
+is provided should be small compared to the total download time so it likely won't be
+something users notice.
+
+Introducing a new authentication system always involves some risk that it will be implemented
+incorrectly, but this addition should be easy enough to understand and address any issues that
+could come up.
+
+## Design Details [optional]
+
+### Open Questions
+
+* Should we switch the on-prem behavior to use the same auth strategy?
+  - Currently when we deploy using the operator we also create a public/private key pair
+    which are used to create auth tokens.
+  - These tokens don't expire and are primarily used to authenticate communication from
+    the agent to assisted service.
+* Should we also support RHSSO user tokens in a header for these endpoints?
+  - The image service could also pass a header through the assisted service
+
+### UI Impact
+
+### Test Plan
+
+- Image downloads from the cloud with the image service and the new authentication work correctly.
+- An image can not be downloaded without a secret or with an incorrect secret.
+- The refresh image secret endpoint invalidates the old token (requests no longer succeed with the old URL) and replaces the URL with a valid one
+- URLs with an existing token expire after 4 hours
+
+## Drawbacks
+
+Implementing our own authentication system is generally risky. We already have a system
+(RHSSO) that works in the cloud, gives us authorization as well, and we don't maintain.
+The only reason we can't use this system for this use case is that we want to make
+the download easier for users. The most secure, and most flexible option would be
+to require an OCM JWT token for this download, just as we do for every other API call.
+
+## Alternatives
+
+* Mint JWTs in our service which we then verify in the request.
+  - This provides us a place to put context for the token (username, expiration, etc),
+    but will require us to maintain our own private and public keys and build out
+    the ability to rotate those keys periodically without breaking existing downlaod URLS.
+  - This is effectively what we do in the operator case, but with the added work of having
+    to cleanly rotate the keys.
+
+* Use RHSSO and a token in the header as we do for all other API requests
```

Comment:
personally it sounds like the right thing to do, is UI our limitations for this one?

Reply:
The UI and BMCs don't know how to pass auth headers

Reply:
Sorry I just saw that you also responded to this @avishayt :sweat_smile: 

I'm not sure BMC is relevant for this as it would only be used in the cloud and we won't support serving directly to a virtual media interface in any case.

Reply:
> The UI and BMCs don't know how to pass auth headers

Are we expecting users to paste the image download URL inside the BMC directly? Right now, users are likely not doing this since the S3 URLs are massive and the BMC can't handle them. Am I missing something?